### ISSUE TITLE
Robotic voice using transfer learning, How to improve it?

### ISSUE THREAD

**Issue by aishweta** [NVIDIA/tacotron2#373]
@rafaelvalle @pravn @ksaidin @CookiePPP

I have American English accents female 7.4 hours of audio dataset. I've removed start and end silence using @Yeongtae  [yeongtae's processing](https://github.com/Yeongtae/tacotron2/blob/monitoring_metric/preprocess_dataset.py)

I've trained data using transfer learning using nvdia-tacotron2 pre-tarined model. I got good alignment plot at 30100 steps. 
For audio synthesis I used pre-trained [waveglow_256channels.pt](https://drive.google.com/u/0/uc?export=download&confirm=VilB&id=1WsibBTsuRg_SF2Z6L6NFRTT-NjEy1oTx)

Questions:
1. Audio sounds robotic, how can I improve voice quality. (I want more natural sound)
2. How to eliminate background noise, without affecting voice quality.

Any comments?

Please check [Audio samples](https://soundcloud.com/shweta-ai/tacotron2-voice)

![30 1](https://user-images.githubusercontent.com/31302697/84569759-7a860000-ada6-11ea-8580-890d610a8286.png)

**Comment by ksaidin** [comment]
Your dataset passed the proof of concept part, now you train from scratch (both tacotron and waveglow) to get better results.

**Comment by aishweta** [comment]
@ksaidin I've very less dataset only 7 hours, as per my understanding I need more than 20 hours of data from scratch training. is that right?

**Comment by ksaidin** [comment]
There is no strict 20 hrs rule. You can give it a try, there are various methods to boost your dataset or penalize model while training if needed.

**Comment by aishweta** [comment]
is training waveglow will improve my results?

**Comment by rafaelvalle** [comment]
The samples you shared sound really good!
You can use [iZotope RX](https://www.izotope.com/en/products/rx.html) to remove the background noise. I assume this background noise is in your training data otherwise it wouldn't be present during inference. Hence, remove the background noise from all samples and train again.

You can change Waveglow's Denoiser denoising strength to get rid of the high whistle like sound.
Also make sure you're using the [latest waveglow waveglow_256channels_universal_v5](https://drive.google.com/file/d/1rpK8CzAAirq9sWZhe9nlfvxMF1dRgFbF/view).

**Comment by aishweta** [comment]
@rafaelvalle Thanks for the reply.

Yes, I've some background noise in my data. I tried to remove noise using audacity, new data hopefully, I'll train in next week.
Btw Thanks for suggesting the approach to eliminate the noise, I'll try. 

I used waveglow's denoiser by increasing and decreasing sigma values, I haven't seen much difference. audios still sound robotic.

**Comment by ksaidin** [comment]
> I tried to use waveglow waveglow_256channels_universal_v5 latest pre-trained model, I was getting some error: "'WN' object has no attribute 'cond_layer'"

[Update the Tacotron2/waveglow](https://github.com/NVIDIA/tacotron2/issues/333#issuecomment-613783118)

**Comment by aishweta** [comment]
@ksaidin I tried this and I changed code glow.py and convert_model.py but I got nan values after waveglow. 

But I've values in mel_postnets

**Comment by ksaidin** [comment]
@shwetagargade216 instead of changing code of glow.py, just update whole "tacotron2/waveglow" from https://github.com/NVIDIA/waveglow/

**Comment by luvwinnie** [comment]
@shwetagargade216 Hi does you resolve the problem of robotic voice? I'm facing the same problem...

**Comment by aishweta** [comment]
@luvwinnie yes my problem has resolved. I have changed the sampling rate using 22050 to 8000 using ffmg. along with it I have dependent parameters and trained both Tacotron2 and waveglow. I go better results than before.

**Comment by luvwinnie** [comment]
@shwetagargade216 Thank you so much! So from your experience, maybe means that we suppose to use lower frequency of data to train the model instead 22050Hz or even 41kHz?
